File: Segundo Periodo/Arquitetura de computadores/2023-2_arq1_guia_01/Python/Dec2Base.py
class Dec2Base:
    def __init__(self):
        self.recive = ''
        self.number = ''
        self.op = 0

    def push_black(self,item):
        self.number = item + self.number

    def converter(self):
        numb = int(self.recive)
        if self.op == 16:
            while numb > 0:
                resto = numb % self.op
                numb //= self.op
                if resto == 10:
                    self.push_black('a')
                elif resto == 11:
                    self.push_black('b')
                elif resto == 11:
                    self.push_black('c')
                elif resto == 11:
                    self.push_black('d')
                elif resto == 11:
                    self.push_black('e')
                else:
                    self.push_black('f')
        else:
            while numb > 0:
                self.number = str(numb % self.op) + self.number
                numb //= self.op

    def getNumber(self):
        return self.number
    
    def isNumber(self,comp):
        for i in comp:
            if not('0' <= i and i <= '9'):
                return False
        return True

    def baseValid(self,base):
        if base in [2,8,16]:
            return True

    def setNumber(self,num,base):
        if base in [2,8,16] and self.isNumber(num):
            self.recive = num
            self.op = base
            self.number = ''
            self.converter()

File: Segundo Periodo/Arquitetura de computadores/2023-2_arq1_guia_01/Java/Dec2Base.java
package Java;

import java.util.Scanner;

public class Dec2Base {
    private String recive = "";
    private String number = "";
    private int base = 0;

    public boolean isNumber(String testar){
        int len = testar.length();
        for(int i = 0; i < len; i++){
            if(!('0' <= testar.charAt(i) && testar.charAt(i) <= '9')){
                return false;
            }
        }
        return true;
    }

    public int toInt(String testar){
        int len = testar.length();
        int temp = 0;
        for(int i = 0; i < len; i++){
            temp += (testar.charAt(i) - '0') * Math.pow(10,len-i-1);
        }
        return temp;
    }

    public void convert(){
        int temp = toInt(this.recive);
        if(this.base == 16){
            while(temp > 0){
                int resto = temp % base;
                temp /= base;
                if(resto > 9){
                    switch(resto){
                        case 10:
                            this.number = 'a' + this.number;
                        case 11:
                            this.number = 'b' + this.number;
                        case 12:
                            this.number = 'c' + this.number;
                        case 13:
                            this.number = 'd' + this.number;
                        case 14:
                            this.number = 'e' + this.number;
                        case 15:
                            this.number = 'f' + this.number;
                    }
                }else{
                    this.number = resto + this.number;
                }
            }
        }else{
            while(temp > 0){
                this.number = (temp % base) + this.number;
                temp /= base;
            }
        }
    }

    public String getNumber(){
        return this.number;
    }

    public void setNumber(String num, int base){
        if ((base != 2 && base != 8 && base != 16) && isNumber(num)){
            System.out.printf("\n\nBase %d invalida!\n\n",base);
        }else{
            this.recive = num;
            this.base = base;
            this.number = "";
            convert();
        }
    }

    public boolean compString(String a,String b){
        int lenA = a.length(), lenB = b.length();
        if(lenA == lenB){
            for(int i = 0; i < lenA; i++){
                if(a.charAt(i) != b.charAt(i)){
                    return false;
                }
            }
        }else{
            return false;
        }
        return true;
    }

    public void executeLoop(){
        String option = "";
        Scanner scanner = new Scanner(System.in);
        while(!compString(option, "end")){
            System.out.print("\nDigite end para sair do loop\nNumero em decimal: ");
            option = scanner.nextLine();
            System.out.println("Precione ENTER!");
            scanner.nextLine();

            if(compString(option, "end")){
                continue;
            }
            System.out.print("Base: ");
            int op = scanner.nextInt();
            scanner.nextLine();
            setNumber(option, op);
            System.out.println(getNumber());
        }
    }

}
